include/cache.hpp

```diff
@@ -67,6 +67,22 @@ class fixed_sized_cache
         }
     }
 
+    std::unique_ptr<Value> TryGet(const Key &key) const
+    {
+        std::unique_ptr<Value> ret;
+
+        read_guard lock{safe_op};
+        auto elem_it = FindElem(key);
+
+        if (elem_it != cache_items_map.end())
+        {
+            cache_policy.Touch(key);
+            ret = std::make_unique<Value>(elem_it->second);
+        }
+
+        return ret;
+    }
```

Comment:
why is this better than

```c++
try {
    return std::make_unique<Value>(Get(key));
} catch (const std::range_error &) {
    return NULL;
}
```

Reply:
I've been doing some measurements on this and it seems like the `try {} catch () {}` block prevents some optimizations, so it's been 10% slower in `CanonicalDBG`.

I can write an isolated benchmark to test this.

Reply:
```c++
#include <cstddef>
#include <cstdlib>
#include <iostream>
#include <cache.hpp>
#include <lru_cache_policy.hpp>


int main() {
    caches::fixed_sized_cache<int, int, caches::LRUCachePolicy<int>> int_cache_(1000);

    int sum = 0;

    for (size_t i = 0; i < 10000000; ++i) {
        int val = 0;
        int j = rand() % 10000;
#ifdef GET
        try {
            val = int_cache_.Get(j);
        } catch (const std::range_error&) {}
#else
        if (auto find = int_cache_.TryGet(j))
            val = *find;
#endif

        if (!val) {
            val = rand();
            int_cache_.Put(j, val);
        }

        sum += val;
    }

    std::cout << "done\t" << sum << "\n";
    return 0;
}
```

The `Get` variant gets runs in about 13 seconds, while the `TryGet` variants runs in around 2 seconds

Reply:
Tested in `g++-11` with the compile command `g++ -o test_caches -std=c++17 -O3 -DNDEBUG -Iinclude test_caches.cpp`

Reply:
If this is the case, we should always use the version with pointers and the version with exceptions should be removed.

Reply:
makes sense. I removed `Get` and fixed the unit tests accordingly

Reply:
There is also std::optional, which might make sense in this use case. But unique pointers also work.

Can you rename `TryGet` to `Get`?